premis_event_service/views.py

```diff
@@ -95,91 +96,104 @@ def humanEvent(request, identifier=None):
     )
 
 
-def paginate_entries(request, entries, num_per_page=20):
-    """
-    paginates a set of entries (set of model objects)
-    """
-
-    # create paginator from result set
-    paginator = Paginator(entries, num_per_page)
-    paginated_entries = None
-    # try to resolve the current page
-    try:
-        page = int(request.GET.get('page', '1'))
-    except ValueError:
-        page = 1
-    try:
-        paginated_entries = paginator.page(page)
-    except (EmptyPage, InvalidPage):
-        paginated_entries = paginator.page(paginator.num_pages)
-    # send back the paginated entries
-    return paginated_entries
+def last_page_ordinal(query_set, per_page=20):
+    qa = query_set.order_by('ordinal')[0:20]
+    return qa.last().ordinal
+
+
+def paginate_events(valid, request, per_page=20):
+    total_events = None
+    events = None
+    page = int(request.GET.get('page', 1))
+    offset = None
+    unfiltered = True
+    last_page_ord = per_page+1
+    if any([v for k, v in valid.items() if k != 'min_ordinal']):
+        events = (Event.objects.search(**valid)
+                  .prefetch_related('linking_objects'))
+        total_events = events.count()
+        offset = (page-1)*per_page
+        last_page_ord = last_page_ordinal(events)
+        if 'min_ordinal' in valid and valid['min_ordinal']:
+            events = events.filter(ordinal__lte=valid['min_ordinal'])[:20]
```

Comment:
The ordinal is in descending order, so the values on the current page would be less than the `min_ordinal`.